fe/fe-core/src/main/java/com/starrocks/catalog/Index.java

```diff
@@ -256,55 +252,36 @@ public TOlapTableIndex toThrift() {
-                commonIndexParamKeySet = Collections.emptySet();
-                indexIndexParamKeySet = Arrays.stream(NgramBfIndexParamsKey.values())
-                        .map(e -> e.name().toUpperCase(Locale.ROOT))
-                        .collect(Collectors.toSet());
-                searchIndexParamKeySet = Collections.emptySet();
-            } else {
-                commonIndexParamKeySet = Collections.emptySet();
-                indexIndexParamKeySet = Collections.emptySet();
-                searchIndexParamKeySet = Collections.emptySet();
-            }
 
+            IndexParams indexParams = IndexParams.getInstance();
+            Map<String, IndexParamItem> commonIndexParams = indexParams.getKeySet(indexType, IndexParamType.COMMON);
+            Map<String, IndexParamItem> indexIndexParams = indexParams.getKeySet(indexType, IndexParamType.INDEX);
+            Map<String, IndexParamItem> searchIndexParams = indexParams.getKeySet(indexType, IndexParamType.SEARCH);
             for (Entry<String, String> propEntry : properties.entrySet()) {
                 String key = propEntry.getKey();
                 String value = propEntry.getValue();
                 String upperKey = key.toUpperCase(Locale.ROOT);
-                if (commonIndexParamKeySet.contains(upperKey)) {
+                if (commonIndexParams.containsKey(upperKey)) {
                     commonProperties.put(key, value);
-                } else if (indexIndexParamKeySet.contains(upperKey)) {
+                    commonIndexParams.remove(upperKey);
+                } else if (indexIndexParams.containsKey(upperKey)) {
                     indexProperties.put(key, value);
-                } else if (searchIndexParamKeySet.contains(upperKey)) {
+                    indexIndexParams.remove(upperKey);
+                } else if (searchIndexParams.containsKey(upperKey)) {
                     searchProperties.put(key, value);
+                    searchIndexParams.remove(upperKey);
                 } else {
                     extraProperties.put(key, value);
                 }
             }
 
-            Arrays.stream(CommonIndexParamKey.values())
-                    .filter(k -> !commonProperties.containsKey(k.name().toLowerCase(Locale.ROOT)) && k.needDefault())
-                    .forEach(k -> commonProperties.put(k.name().toLowerCase(Locale.ROOT), k.defaultValue()));
-            Arrays.stream(IndexParamsKey.values())
-                    .filter(k -> !indexProperties.containsKey(k.name().toLowerCase(Locale.ROOT)) && k.needDefault())
-                    .forEach(k -> indexProperties.put(k.name().toLowerCase(Locale.ROOT), k.defaultValue()));
-            Arrays.stream(SearchParamsKey.values())
-                    .filter(k -> !searchProperties.containsKey(k.name().toLowerCase(Locale.ROOT)) && k.needDefault())
-                    .forEach(k -> searchProperties.put(k.name().toLowerCase(Locale.ROOT), k.defaultValue()));
+            // set default value
```

Comment:
actually i do not recommend to set default value by toThrfit(), in this way, show index from table will not show default values. A better way is add to properties when analyze

Reply:
Ok, I'm on it, I'll do default value setting when analyze